src/wasm/wasm-binary.cpp

```diff
@@ -795,6 +795,33 @@ void WasmBinaryWriter::writeNames() {
   // TODO: label, type, and element names
   // see: https://github.com/WebAssembly/extended-name-section
 
+  // GC field names
+  if (wasm->features.hasGC()) {
+    std::vector<HeapType> namedTypes;
+    for (auto& type : types) {
+      if (type.isStruct() && wasm->typeNames.count(type) &&
+          !wasm->typeNames.at(type).fieldNames.empty()) {
+        namedTypes.push_back(type);
+      }
+    }
+    if (!namedTypes.empty()) {
+      auto substart =
+        startSubsection(BinaryConsts::UserSections::Subsection::NameField);
+      o << U32LEB(namedTypes.size());
+      for (Index i = 0; i < namedTypes.size(); i++) {
+        auto type = namedTypes[i];
+        o << U32LEB(i);
```

Comment:
IIUC, this emits the index into just the types with named fields, but it should emit the index into all the defined types.

Reply:
Good catch!

Reply:
Updated. This fixed something in the test suite, that I missed while scanning the output... I verified it carefully now.